src/main/java/duke/internal/Storage.java

```diff
@@ -0,0 +1,56 @@
+package duke.internal;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * Handles the storage of the task list, reading and/or writing to a file.
+ */
+public class Storage {
+    private final Path path;
+
+    public Storage(Path path) {
+        this.path = path;
+    }
+
+    /**
+     * Returns a new TaskList from a file of serialized tasks.
+     *
+     * @return the new TaskList
+     * @throws IOException if an I/O error occurs reading the file
+     * @since Level-7
+     */
+    public TaskList load() throws IOException {
+        List<Task> tasks = Files.readAllLines(path)
+                .stream()
+                .map(Task::deserialize)
+                .collect(Collectors.toList());
+        return new TaskList(tasks);
+    }
+
+    /**
+     * Writes the list of tasks to a file at the given path.
+     * If the file already exists, it will be overwritten.
+     *
+     * @throws IOException if an I/O error occurs writing to the file
+     * @since Level-7
+     */
+    public void save(TaskList tasks) throws IOException {
```

Comment:
could this be called saveTask and loadTask instead?

Reply:
Good suggestion! I'll amend this in the next iteration.